packages/core/src/app/address/googleAutocomplete/GoogleAutocomplete.tsx

```diff
@@ -21,53 +21,46 @@ export interface GoogleAutocompleteProps {
     onChange?(value: string, isOpen: boolean): void;
 }
 
-interface GoogleAutocompleteState {
-    items: AutocompleteItem[];
-    autoComplete: string;
-}
-
-class GoogleAutocomplete extends PureComponent<GoogleAutocompleteProps, GoogleAutocompleteState> {
-    googleAutocompleteService: GoogleAutocompleteService;
-
-    constructor(props: GoogleAutocompleteProps) {
-        super(props);
-        this.googleAutocompleteService = new GoogleAutocompleteService(props.apiKey);
-        this.state = {
-            items: [],
-            autoComplete: 'off',
-        };
-    }
-
-    render(): ReactNode {
-        const { initialValue, onToggleOpen = noop, inputProps = {} } = this.props;
-
-        const { autoComplete, items } = this.state;
-
-        return (
-            <Autocomplete
-                defaultHighlightedIndex={-1}
-                initialHighlightedIndex={-1}
-                initialValue={initialValue}
-                inputProps={{
-                    ...inputProps,
-                    autoComplete,
-                }}
-                items={items}
-                listTestId="address-autocomplete-suggestions"
-                onChange={this.onChange}
-                onSelect={this.onSelect}
-                onToggleOpen={onToggleOpen}
-            >
-                <div className="co-googleAutocomplete-footer" />
-            </Autocomplete>
-        );
-    }
-
-    private onSelect: (item: AutocompleteItem) => void = (item) => {
-        const { fields, onSelect = noop, nextElement } = this.props;
-
-        this.googleAutocompleteService.getPlacesServices().then((service) => {
-            service.getDetails(
+const toAutocompleteItems = (
+    results?: google.maps.places.AutocompletePrediction[],
+): AutocompleteItem[] => {
+    return (results || []).map((result) => ({
+        label: result.description,
+        value: result.structured_formatting.main_text,
+        highlightedSlices: result.matched_substrings,
+        id: result.place_id,
+    }));
+};
+
+const GoogleAutocomplete: React.FC<GoogleAutocompleteProps> = ({
+    initialValue,
+    onToggleOpen = noop,
+    inputProps = {},
+    fields,
+    onSelect = noop,
+    nextElement,
+    isAutocompleteEnabled,
+    onChange = noop,
+    componentRestrictions,
+    types,
+    apiKey,
+}) => {
+    const [items, setItems] = useState<AutocompleteItem[]>([]);
+    const [autoComplete, setAutoComplete] = useState<string>('off');
+    const googleAutocompleteServiceRef = useRef<GoogleAutocompleteService>();
```

Comment:
Do we need to create a ref here? Feel we can just create an instance when the component mounts?